tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala

```diff
@@ -145,18 +157,29 @@ class Profiler(hadoopConf: Configuration, appArgs: ProfileArgs) extends Logging
 
     class ProfileThread(path: EventLogInfo, index: Int) extends Runnable {
       def run: Unit = {
-        val appOpt = createApp(path, numOutputRows, index, hadoopConf)
-        appOpt.foreach { app =>
-          val sum = try {
-            val (s, _) = processApps(Seq(app), false, profileOutputWriter)
-            Some(s)
-          } catch {
-            case e: Exception =>
-              logWarning(s"Unexpected exception thrown ${path.eventLog.toString}, skipping! ", e)
-              None
-
+        try {
+          val appOpt = createApp(path, numOutputRows, index, hadoopConf)
+          appOpt.foreach { app =>
+            val sum = try {
+              val (s, _) = processApps(Seq(app), false, profileOutputWriter)
+              Some(s)
+            } catch {
+              case e: Exception =>
+                logWarning(s"Unexpected exception thrown ${path.eventLog.toString}, skipping! ", e)
+                None
+            }
+            sum.foreach(allApps.add(_))
           }
-          sum.foreach(allApps.add(_))
+        } catch {
+          case oom: OutOfMemoryError =>
+            logError(s"OOM error while processing large file: ${path.eventLog.toString}." +
+                s"Increase heap size.", oom)
+            System.exit(1)
+          case o: Error =>
+            logError(s"Error occured while processing file: ${path.eventLog.toString}", o)
+            System.exit(1)
+          case e: Exception =>
+            logWarning(s"Exception occurred processing file: ${path.eventLog.getName}", e)
```

Comment:
nit: we could define a recover partial function instead of copy-and-paste.
and use in `Try {  ... }.recover/With(recoverFunc)`